File: clients/go/examples/basic_crawl.go
package main

import (
	"context"
	"fmt"
	"log"

	crawlkit "github.com/WyattAu/crawlkit/clients/go"
)

func main() {
	client := crawlkit.NewClient("http://localhost:4000", "your-api-key")
	ctx := context.Background()

	// Start a crawl
	resp, err := client.StartCrawl(ctx, crawlkit.CrawlRequest{
		StartURL:    "https://example.com",
		MaxPages:    50,
		Concurrency: 4,
	})
	if err != nil {
		log.Fatal(err)
	}
	fmt.Printf("Crawl started: %s\n", resp.CrawlID)

	// Get crawl status
	status, err := client.GetCrawl(ctx, resp.CrawlID)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Printf("Status: %s\n", status.Status)

	// Get stats
	stats, err := client.GetCrawlStats(ctx, resp.CrawlID)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Printf("Pages crawled: %d\n", stats.TotalPages)
	fmt.Printf("Issues found: %d\n", stats.TotalIssues)

	// List users
	users, err := client.ListUsers(ctx)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Printf("Users: %d\n", len(users))
}

File: clients/nodejs/examples/basic_crawl.ts
import { CrawlkitClient, CrawlRequest } from "../src";

async function main() {
  const client = new CrawlkitClient("http://localhost:4000", "your-api-key");

  // Start a crawl
  const req: CrawlRequest = {
    start_url: "https://example.com",
    max_pages: 50,
    concurrency: 4,
  };

  const resp = await client.startCrawl(req);
  console.log(`Crawl started: ${resp.crawl_id}`);

  // Get crawl status
  const status = await client.getCrawl(resp.crawl_id);
  console.log(`Status: ${status.status}`);

  // Get stats
  const stats = await client.getCrawlStats(resp.crawl_id);
  console.log(`Pages crawled: ${stats.total_pages}`);
  console.log(`Issues found: ${stats.total_issues}`);

  // List users
  const users = await client.listUsers();
  console.log(`Users: ${users.length}`);
}

main().catch(console.error);
